### PR TITLE
Refactor HiveConnector.h to cleanup includes

### PR DESCRIPTION
**Pull request by majetideepak**
Includes in public header files can result in header leaks and result in unnecessary dependencies 
on the clients. This change removes Protobuf header requirement in Prestissimo.

### CODE REVIEW

**Comment by majetideepak** [comment]
Presto side changes are here: https://github.com/prestodb/presto/pull/20370

File: velox/connectors/hive/HiveConnector.cpp [hunk 1]
```diff
@@ -62,6 +65,44 @@ HiveConnector::HiveConnector(
           std::make_unique<FileHandleGenerator>(std::move(properties))),
       executor_(executor) {}
 
+std::unique_ptr<DataSource> HiveConnector::createDataSource(
```
Comment (kgpai) [inline velox/connectors/hive/HiveConnector.cpp:68]: I dont see this being used anywhere in this PR . Anything basic I am missing ?

File: velox/connectors/hive/HiveConnector.cpp [hunk 2]
```diff
@@ -62,6 +65,44 @@ HiveConnector::HiveConnector(
...
+      &fileHandleFactory_,
+      connectorQueryCtx->expressionEvaluator(),
+      connectorQueryCtx->allocator(),
+      connectorQueryCtx->scanId(),
+      HiveConfig::isFileColumnNamesReadAsLowerCase(connectorQueryCtx->config()),
+      executor_,
+      options);
+}
+
+std::unique_ptr<DataSink> HiveConnector::createDataSink(
```
Comment (kgpai) [inline velox/connectors/hive/HiveConnector.cpp:93]: Same here.

**Review by kgpai — COMMENTED** [review]
Hi @majetideepak , I am curious how you are validating that this change removes the leak in protobuf headers. Thanks for doing this btw .

File: velox/connectors/hive/HiveConnector.cpp (hunk 1 above)
Comment (majetideepak) [reply velox/connectors/hive/HiveConnector.cpp:68]: This is just being moved from the `HiveConnector.h` header to the cpp file.

File: velox/connectors/hive/HiveConnector.cpp (hunk 2 above)
Comment (majetideepak) [reply velox/connectors/hive/HiveConnector.cpp:93]: This is just being moved from the `HiveConnector.h` header to the cpp file.

**Comment by majetideepak** [comment]
@kgpai the high-level idea is that in a library like Velox, all public headers such as `HiveConnector.h` must include other headers minimally. This improves build time and also avoids header creep.
The changes here are simply removing the other includes in `HiveConnector.h` which is a public header.

**Comment by kgpai** [comment]
@majetideepak Thanks for the explanation .

**Review by kgpai — APPROVED** [review]

**Review by xiaoxmeng — APPROVED** [review]
@majetideepak thanks for the refactoring!

**Comment by kgpai** [comment]
@majetideepak This change requires more changes internally - hoping to finish those soon and land this latest tomorrow. Appreciate your patience.

**Comment by majetideepak** [comment]
@kgpai Any updates on this? Thanks.

**Pull request closed without merging**